components/AlgoliaSearch.js

```diff
@@ -0,0 +1,219 @@
+import { useState, useEffect, useCallback, useRef, createContext, useContext } from 'react'
+import { createPortal } from 'react-dom'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+import Head from 'next/head'
+import { DocSearchModal } from '@docsearch/react'
+import clsx from 'clsx'
+
+const INDEX_NAME = 'asyncapi';
+const APP_ID = 'Z621OGRI9Y';
+const API_KEY = '09fe8dc96d776bfaf2da47a6d5f57a3f';
+
```

Comment:
Since the key is expected to be here, I would probably add a comment clarifying it 🙂